### PR TITLE
Finish Collaboration Development Index Metrics Model

### PR DESCRIPTION
**Pull request by YangWenhao3906**

### CODE REVIEW

File: compass_common/math_utils.py [hunk 1]
```diff
@@ -1,11 +0,0 @@
-def get_medium(L):
```
Comment (eyehwan) [inline compass_common/math_utils.py:1]: Why do you move it to compass_common/model_score_algorithm_utils.py?

File: compass_metrics/db_dsl.py [hunk 2]
```diff
@@ -305,4 +305,125 @@ def get_pr_closed_uuid_count(option, repos_list, field, from_date=str_to_datetim
         }
     }
 
-    return query
\ No newline at end of file
+    return query
+
+
+def get_pr_message_count(repos_list, field, date_field="grimoire_creation_date", size=0, filter_field=None, from_date=str_to_datetime("1970-01-01"), to_date=datetime_utcnow()):
```
Comment (eyehwan) [inline compass_metrics/db_dsl.py:311]: better adding docs.

**Review by eyehwan — COMMENTED** [review]
@lishengbao because of your "bad" example :) , I didn't see enough docs for methods, especially for metrics methods. So I suggest adding docs refered from compass docs, for example: https://oss-compass.org/docs/metrics-models/productivity/collaboration-development-index/#commit-frequency.

**Comment by lishengbao** [comment]
LGTM

**Comment by lishengbao** [comment]
> @lishengbao because of your "bad" example :) , I didn't see enough docs for methods, especially for metrics methods. So I suggest adding docs refered from compass docs, for example: https://oss-compass.org/docs/metrics-models/productivity/collaboration-development-index/#commit-frequency.

I've filled in the relevant docs later.

**Comment by YangWenhao3906** [comment]
> Why do you move it to compass_common/model_score_algorithm_utils.py? 

I am considering organizing all the math-related components into a single module. However, if I proceed with this approach, it would be necessary to rename the file `model_score_algorithm_utils`.

Of course, keeping them separated in two modules may result in lower coupling.

What do you think? Need I change it back? Or we give this module another name like "math_utils"?

**Comment by YangWenhao3906** [comment]
> > @lishengbao because of your "bad" example :) , I didn't see enough docs for methods, especially for metrics methods. So I suggest adding docs refered from compass docs, for example: https://oss-compass.org/docs/metrics-models/productivity/collaboration-development-index/#commit-frequency.

I've filled in the relevant docs later. 

thx~

**Comment by eyehwan** [comment]
> What do you think? Need I change it back? Or we give this module another name like "math_utils"?

managed into one module is fine with one, how about call it algorithm_utils? @YangWenhao3906 @lishengbao

**Comment by lishengbao** [comment]
> managed into one module is fine with one, how about call it algorithm_utils? @YangWenhao3906 @lishengbao
agreed

**Review by eyehwan — APPROVED** [review]

**Comment by YangWenhao3906** [comment]
#52

**Pull request merged**